cmd/version.go

```diff
@@ -0,0 +1,79 @@
+package cmd
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	"github.com/spf13/cobra"
+)
+
+// GitCommit is the short git commit hash from the environment
+var GitCommit string
+
+// Version is the tag version from the environment
+var Version string
+
+// githubResponse is a necessary struct for the JSON unmarshalling that is happening
+// in the getLatestVersion().
+type gitHubResponse struct {
+	TagName string `json:"tag_name"`
+}
+
+type versionResponse struct {
+	Commit  string `json:"commit`
+	Version string `json:"version`
+	Latest  string `json:"latest`
+}
+
+var versionCmd = &cobra.Command{
+	Use:   "version",
+	Short: "Display the version",
+	Long:  "Display version of osdctl",
+	Run:   version,
```

Comment:
this can be changed to RunE which does allow to return an error
so maybe that solves your problem

Reply:
Yes it does. Thank you!